### ISSUE TITLE
[Feature] Support using Models as pivot tables

### ISSUE THREAD

**Issue by tabacitu** [Laravel-Backpack/ideas#195]
# Bug report

### What I did

Added Hero to the demo, which _should_ be using `monsters` as the pivot table. The idea was that... if Monster were a pivot table... we would be able to test ALL field types, if they work as pivot fields.

That would help us fill in the last column in [this table](https://github.com/Laravel-Backpack/CRUD/issues/4109#issuecomment-1019323850).

### What I expected to happen

Create / Update pages to load.

### What happened

![Screenshot 2022-02-02 at 20 19 32](https://user-images.githubusercontent.com/1032474/152213944-f30cf0f5-8afa-4ef6-b26b-cb5652a1516d.png)

### What I've already tried to fix it

Nothing yet, since this is not exactly urgent. We can add this later, as a feature. I think what fields we support right now as pivot subfields are already MORE than enough for most projects.

Basically, what I think we need to do is:
- create a new MonsterPivot model, that [extends `Illuminate\Database\Eloquent\Relations\Pivot`](https://laravel.com/docs/8.x/eloquent-relationships#defining-custom-intermediate-table-models).. or better yet, just has those few methods and properties on it and extends Monster;
- make sure Monster has all relationships it needs;
- test all relationship fields, that they work as pivot subfields; 

### Is it a bug in the latest version of Backpack?

v4.2/v5 - whatever you wanna call it
